document/src/main/java/com.google.android.fhir.document/utils/GenerateShlUtils.kt

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.document.utils
+
+import android.content.ContentValues.TAG
+import android.content.Context
+import android.graphics.Bitmap
+import android.os.Handler
+import android.os.Looper
+import android.util.Base64
+import android.widget.ImageView
+import ca.uhn.fhir.context.FhirContext
+import ca.uhn.fhir.context.FhirVersionEnum
+import com.google.android.fhir.document.dataClasses.SHLData
+import com.google.android.fhir.document.interfaces.RetrofitSHLService
+import java.lang.Exception
+import java.text.SimpleDateFormat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.json.JSONObject
+import timber.log.Timber
+
+internal class GenerateShlUtils(
```

Comment:
suggestion (and this is what we've been slowly doing with the refactors): let's now separate the qrGeneration from this class. What I suggest is that this class becomes an implementation of `SHLinkGenerator`, so you'd get rid of any code that is related to QR generation. 

```suggestion
internal class SHLinkGeneratorImpl(
```